Fix typos and adds some more texts (#2661)
airbytehq/airbyte: docs/architecture/incremental-append.md

## Known Limitations

> Due to the use of a cursor column, if modifications to the underlying records are made without properly updating the cursor field, then the updated records won't be picked up by the **Incremental** sync as expected because the source connectors is extracting delta rows using a SQL query looking like:

Reviewer: ```suggestion
Due to the use of a cursor column, if modifications to the underlying records are made without properly updating the cursor field, then the updated records won't be picked up by the **Incremental** sync as expected since the source connectors extract delta rows using a SQL query looking like:
```

]
```

> In the next sync, the delta applied to the source data contains the following record:

Reviewer: ```suggestion
At the start of the next sync, the source data contains the following new record:
```

```javascript
[
    { "name": "Louis XVI", "deceased": true, "updated_at":  1754 },
]
```

> At the end of this incremental sync, the data warehouse would still contain data from the first sync because the delta record did not provide a valid value for the cursor field (the cursor field is not greater than last sync's max value, `1754 < 1755`), so it is not emitted by the source as a new or modified record.

Reviewer: ```suggestion
At the end of the second incremental sync, the data warehouse would still contain data from the first sync because the delta record did not provide a valid value for the cursor field (the cursor field is not greater than last sync's max value, `1754 < 1755`), so it is not emitted by the source as a new or modified record.
```

Similarly, if multiple modifications are made during the same day to the same records.
If the frequency of the sync is not granular enough (for example, set for every 24h),

> then intermediate modifications to the data is not going to be detected and emitted.

Reviewer: ```suggestion
then intermediate modifications to the data are not going to be detected and emitted.
```

> Only the state of the data of when the sync is running will be reflected in the destination.

Reviewer: ```suggestion
Only the state of the data at the time the sync runs will be reflected in the destination.
```

Those concerns could be solved by using a different sync mode based on binary logs, Write-Ahead-Logs \(WAL\), or also called **Incremental - Change Data Capture**. \(coming to Airbyte in the near future\).

> The current behavior of **Incremental** is not able to handle when the source's schema changes, for example, when a column is added, renamed or deleted to an existing stream,such events yet. It is recommended to trigger a [Full refresh - Overwrite](full-refresh-overwrite.md) to correctly replicate the data to the destination with the new schema changes.

Reviewer: ```suggestion
The current behavior of **Incremental** is not able to handle source schema changes yet, for example, when a column is added, renamed or deleted from an existing table etc. It is recommended to trigger a [Full refresh - Overwrite](full-refresh-overwrite.md) to correctly replicate the data to the destination with the new schema changes.
```

> When the source's schema changes, for example, when a column is added, renamed or deleted to an existing stream, the current behavior of **Incremental** is not able to handle such events yet. Therefore, it is recommended to trigger a [Full refresh - Overwrite](full-refresh-overwrite.md) to recreate at the destination the data with the new metadata included.

Reviewer: ```suggestion
The current behavior of **Incremental** is not able to handle when the source's schema changes, for example, when a column is added, renamed or deleted to an existing stream,such events yet. It is recommended to trigger a [Full refresh - Overwrite](full-refresh-overwrite.md) to recreate at the destination the data with the new metadata included.
```
